bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapJSonDTO.java

```diff
@@ -110,14 +120,13 @@ public CoIotGenericSensorList read(final JsonReader in) throws IOException {
 
             in.beginObject();
             String generic = in.nextName();
-            Validate.notNull(generic, "Invalid JSon format for CoIotSensorList");
             if (generic.equals(COIOT_TAG_GENERIC)) {
                 in.beginArray();
                 while (in.hasNext()) {
                     in.beginArray();
                     final CoIotSensor sensor = new CoIotSensor();
                     in.nextInt(); // alway 0
-                    sensor.index = new Integer(in.nextInt()).toString();
+                    sensor.id = new Integer(in.nextInt()).toString();
```

Comment:
No need to create an Integer instance when you don't have to.
```suggestion
                    sensor.id = Integer.toString(in.nextInt());
```

Reply:
changed